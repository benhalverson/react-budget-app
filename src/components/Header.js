import React from 'react';
import { NavLink } from 'react-router-dom';
const Header = () => (
  <header>
    <h1>Budget app!</h1>
    <nav>
      <ul>
        <li><NavLink to="/" activeClassName="is-active" exact={true}>Login</NavLink></li>
        <li><NavLink to="/dashboard" activeClassName="is-active">Dashboard</NavLink></li>
        <li><NavLink to="/create" activeClassName="is-active">Create Expense</NavLink></li>
        <li><NavLink to="/help" activeClassName="is-active">Help</NavLink></li>
      </ul>
    </nav>
  </header>
)

export default Header;