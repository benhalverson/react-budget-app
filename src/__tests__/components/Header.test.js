import React from 'react';
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json';
import Header from '../../components/Header';

xtest('should render header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});


xtest('should call startLogout on button click', () => {
  const startLogout = jest.fn();
  const wrapper = shallow(<Header startLogout={startLogout} />);
  wrapper.find('button').simulate('click');
  expect(startLogout).toHaveBeenCalled();
})
