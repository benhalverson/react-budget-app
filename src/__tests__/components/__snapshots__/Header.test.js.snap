// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    Budget app!
  </h1>
  <nav>
    <ul>
      <li>
        <NavLink
          activeClassName="is-active"
          aria-current="page"
          exact={true}
          to="/"
        >
          Login
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="is-active"
          aria-current="page"
          to="/dashboard"
        >
          Dashboard
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="is-active"
          aria-current="page"
          to="/create"
        >
          Create Expense
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="is-active"
          aria-current="page"
          to="/help"
        >
          Help
        </NavLink>
      </li>
    </ul>
  </nav>
</header>
`;
